src/App.jsx

```diff
@@ -14,45 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_TYPE = { length: 'length', alphabet: 'alphabet' };
+
+function sortPreparedGoods(arr, sortBy, isReverse) {
+  let preparedGoodsList = [...arr];
+
+  switch (sortBy) {
+    case SORT_TYPE.length:
+      preparedGoodsList.sort((a, b) => a.length - b.length);
+      break;
+
+    case SORT_TYPE.alphabet:
+      preparedGoodsList.sort((a, b) => a.localeCompare(b));
+      break;
+
+    default: preparedGoodsList = [...goodsFromServer];
+  }
```

Comment:
lets take a look at `default: preparedGoodsList = [...goodsFromServer];`

do you have `goodsFromServer`  in `sortPreparedGoods` function?
instead you have `arr` argument...you have to use it

also you have already done the copy `let preparedGoodsList = [...arr];`
and in `default` you do one more copy

is it necessary to do the copy of arr twice?

Reply:
I am sorry, my fault.